cmd/docs.go

```diff
@@ -67,24 +97,145 @@ func docsMain(cmd *cobra.Command, args []string) exitCode {
+	for _, action := range actions {
+		supportedActions.WriteString(fmt.Sprintf("* %v\n", action))
+		var actionDescriptionLines []string
+		switch action {
+		case plugin.ListAction().Name:
+			actionDescriptionLines = []string{
+				fmt.Sprintf("- ls %s", path),
+				fmt.Sprintf("    Type 'docs <child>' to view an ls'ed child's documentation"),
+				fmt.Sprintf("- cd %s", path),
+				fmt.Sprintf("    The 'W' environment variable contains the Wash root. Use 'cd $W' to"),
+				fmt.Sprintf("    return to it."),
+				fmt.Sprintf("- stree %s", path),
+				fmt.Sprintf("    Gives you a high-level overview of the kinds of things that you will"),
+				fmt.Sprintf("    encounter when you 'cd' and 'ls' through this entry."),
+				fmt.Sprintf("- find %s <expression>", path),
+				fmt.Sprintf("    Prints all children that satisfy <expression>. Type 'find --help' to"),
+				fmt.Sprintf("    see all the things that you can filter on."),
+				fmt.Sprintf("- (anything else that works with directories [e.g. 'tree'])"),
+			}
+		case plugin.ReadAction().Name:
+			actionDescriptionLines = []string{
+				fmt.Sprintf("- cat %s", path),
+				fmt.Sprintf("- grep 'foo' %s", path),
+				fmt.Sprintf("- (anything else that reads files [e.g. 'less'])"),
+			}
+		case plugin.StreamAction().Name:
+			actionDescriptionLines = []string{
+				fmt.Sprintf("- tail -f %s", path),
+			}
+		case plugin.WriteAction().Name:
+			if entry.Attributes.HasSize() && entry.Supports(plugin.ReadAction()) {
+				// Entry is file-like so include file-like examples
+				actionDescriptionLines = []string{
+					fmt.Sprintf("- echo 'foo' > %s", path),
+					fmt.Sprintf("    Overwrites the file with 'foo'"),
+					fmt.Sprintf("- echo 'foo' >> %s", path),
+					fmt.Sprintf("    Appends 'foo' to the file"),
+					fmt.Sprintf("- vim %s", path),
+					fmt.Sprintf("    Edits the file with 'vim'"),
+					fmt.Sprintf("- (anything else that lets you write/edit files)"),
+				}
+			} else {
+				actionDescriptionLines = []string{
+					fmt.Sprintf("- echo 'foo' > %s", path),
+					fmt.Sprintf("- echo 'foo' >> %s", path),
+					fmt.Sprintf("    Both commands write the chunk 'foo' to the entry"),
+					fmt.Sprintf("- (anything else that writes files [excluding editors like 'vim'])"),
+				}
+			}
+		case plugin.ExecAction().Name:
+			actionDescriptionLines = []string{
+				fmt.Sprintf("- wexec %s <command> <args...>", path),
+				fmt.Sprintf("    e.g. wexec %s uname", path),
+			}
+		case plugin.DeleteAction().Name:
+			actionDescriptionLines = []string{
+				fmt.Sprintf("- delete %s", path),
+			}
+		case plugin.SignalAction().Name:
+			actionDescriptionLines = []string{
+				fmt.Sprintf("- signal <signal> %s", path),
+				fmt.Sprintf("    e.g. signal start %s", path),
+			}
+		}
+		for _, line := range actionDescriptionLines {
+			supportedActions.WriteString(fmt.Sprintf("    %v\n", line))
+		}
+	}
+	supportedActions.WriteString("\nType 'help' to see a list of all the built-in Wash commands.")
```

Comment:
May be worth mentioning tab-completion and globbing somewhere, but not sure.

Reply:
Hopefully people just try it. I wouldn't add it without more reason.